src/main/scala/org/bitcoins/core/util/Base58.scala

```diff
@@ -0,0 +1,80 @@
+package org.bitcoins.core.util
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success}
+
+/**
+  * Created by chris on 5/16/16.
+  * source of values: https://en.bitcoin.it/wiki/Base58Check_encoding
+  */
+trait Base58 extends BitcoinSLogger {
+
+  val base58Characters = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+  val base58Pairs = base58Characters.zipWithIndex.toMap
+
+
+  def decodeCheck(input: String) : Seq[Byte] = {
```

Comment:
Refactor decodeCheck to have a return type of 'Try[Seq[Byte]]` - this allows us to encode the exception in to the type system instead of just throwing the exception. This is desired behavior because the user expects an exception since the decoding is checked. I'll explain more in person